Feature/lambda versions

ejaros opened the pull request.

Commit:
add lambda aliases to deploy srcipts and to workflow definitions, update tests

Commit:
add missing format string "f"

coreycollins reviewed (changes requested):
A couple things:

- The version string can and probably will be used for more than just lambda function names. The variable name should reflect that. e.g. `sdk_version`

- To go along with that first point, because it can be used in other places, the sdk_version should be an instantiated property on transforms as well as the workflow.

Commit:
use simpler lambda alias for production releases

Commit:
rename lambda_version sdk_version

Commit:
add sdk_version os property on Transform class
 - default sdk_version to v1,
 - update tests

Commit:
remnove unused init args in workflow test

Commit:
refactor Transform and Transform.states()
 - remove _on_fail and set via states function
 - remove sdk_version and set via states function
 - remove _wait_seconds and set at static value 60

Commit:
use arg keywords when calling Transform.states(...) in workflow.py for clarity
